### PR TITLE
Create packaging pipeline to create/test/publish torch-ort packages for n+1 versions of pytorch and onnxruntime

### PR DESCRIPTION
**Pull request by liqunfu**

### CODE REVIEW

File: docker/Dockerfile.ort-default-stable-torch190-onnxruntime181-cu102-cudnn7-devel-ubuntu18.04 [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+# PyTorch
+RUN pip install onnx ninja
+RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION}+${TORCH_CUDA_VERSION} -f https://download.pytorch.org/whl/torch_stable.html
+
+# ORT Module
+RUN pip install onnxruntime-training==${ONNXRUNTIME_TRAINING_VERSION}
+
+RUN pip install torch-ort
+
```
Comment (thiagocrepaldi) [inline docker/Dockerfile.ort-default-stable-torch190-onnxruntime181-cu102-cudnn7-devel-ubuntu18.04:33]:
```suggestion
RUN python -m torch_ort.configure
```

Should we call the PyTorch CPP extension build?

File: docker/Dockerfile.ort-torch-and-onnxruntime-nightly-cu111-cudnn8-devel-ubuntu18.04 [hunk 2]
```diff
@@ -26,14 +19,10 @@ RUN pip install --upgrade pip
...
-# RUN pip install torch==1.8.1+cu111 torchvision==0.9.1+cu111 torchaudio==0.8.1 -f https://download.pytorch.org/whl/torch_stable.html
 RUN pip install --pre torch -f https://download.pytorch.org/whl/nightly/cu111/torch_nightly.html
 
 # ORT Module
-# Uncomment one of the lines below. First one gets stable version. Second gets nightly build.
-# RUN pip install onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_stable_cu111.html
-RUN pip install --pre onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_cu111.html
+RUN pip install --pre onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_torch190.cu111.html
 
 RUN pip install torch-ort
```
Comment (thiagocrepaldi) [inline docker/Dockerfile.ort-torch-and-onnxruntime-nightly-cu111-cudnn8-devel-ubuntu18.04:27]:
```suggestion
RUN pip install torch-ort
RUN python -m torch_ort.configure
```

File: docker/Dockerfile.ort-torch181-onnxruntime-nightly-cu102-cudnn7-devel-ubuntu18.04 [hunk 3]
```diff
@@ -26,14 +24,10 @@ RUN pip install --upgrade pip
...
-# RUN pip install --pre torch -f https://download.pytorch.org/whl/nightly/cu102/torch_nightly.html
+RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION}+${TORCH_CUDA_VERSION} -f https://download.pytorch.org/whl/torch_stable.html
 
 # ORT Module
-# Uncomment one of the lines below. First one gets stable version. Second gets nightly build.
-# RUN pip install onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_stable_cu102.html
-RUN pip install --pre onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_cu102.html
+RUN pip install --pre onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_torch190.cu102.html
 
 RUN pip install torch-ort
```
Comment (thiagocrepaldi) [inline docker/Dockerfile.ort-torch181-onnxruntime-nightly-cu102-cudnn7-devel-ubuntu18.04:32]:
```suggestion
RUN pip install torch-ort
RUN python -m torch_ort.configure

```

File: docker/Dockerfile.ort-torch181-onnxruntime-nightly-cu111-cudnn8-devel-ubuntu18.04 [hunk 4]
```diff
@@ -26,14 +24,10 @@ RUN pip install --upgrade pip
...
-# RUN pip install --pre torch -f https://download.pytorch.org/whl/nightly/cu111/torch_nightly.html
+RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION}+${TORCH_CUDA_VERSION} -f https://download.pytorch.org/whl/torch_stable.html
 
 # ORT Module
-# Uncomment one of the lines below. First one gets stable version. Second gets nightly build.
-# RUN pip install onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_stable_cu111.html
-RUN pip install --pre onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_cu111.html
+RUN pip install --pre onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_torch190.cu111.html
 
 RUN pip install torch-ort
```
Comment (thiagocrepaldi) [inline docker/Dockerfile.ort-torch181-onnxruntime-nightly-cu111-cudnn8-devel-ubuntu18.04:32]:
```suggestion
RUN pip install torch-ort
RUN python -m torch_ort.configure

```

File: docker/Dockerfile.ort-torch181-onnxruntime181-cu102-cudnn7-devel-ubuntu18.04 [hunk 5]
```diff
@@ -26,14 +24,10 @@ RUN pip install --upgrade pip
...
-# RUN pip install --pre torch -f https://download.pytorch.org/whl/nightly/cu102/torch_nightly.html
+RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION}+${TORCH_CUDA_VERSION} -f https://download.pytorch.org/whl/torch_stable.html
 
 # ORT Module
-# Uncomment one of the lines below. First one gets stable version. Second gets nightly build.
-RUN pip install onnxruntime-training==1.8.0
-# RUN pip install --pre onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_cu102.html
+RUN pip install --pre onnxruntime-training==1.8.1 -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_stable_torch181.cu102.html
 
 RUN pip install torch-ort
```
Comment (thiagocrepaldi) [inline docker/Dockerfile.ort-torch181-onnxruntime181-cu102-cudnn7-devel-ubuntu18.04:32]:
```suggestion
RUN pip install torch-ort
RUN python -m torch_ort.configure

```

File: docker/Dockerfile.ort-torch181-onnxruntime181-cu111-cudnn8-devel-ubuntu18.04 [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+# PyTorch
+RUN pip install onnx ninja
+RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION}+${TORCH_CUDA_VERSION} -f https://download.pytorch.org/whl/torch_stable.html
+
+# ORT Module
+RUN pip install onnxruntime-training==1.8.1 -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_stable_torch181.cu111.html
+
+
+RUN pip install torch-ort
+
```
Comment (thiagocrepaldi) [inline docker/Dockerfile.ort-torch181-onnxruntime181-cu111-cudnn8-devel-ubuntu18.04:33]:
```suggestion
RUN python -m torch_ort.configure

```

File: docker/Dockerfile.ort-torch190-onnxruntime-nightly-cu102-cudnn7-devel-ubuntu18.04 [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+# PyTorch
+RUN pip install onnx ninja
+RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION}+${TORCH_CUDA_VERSION} -f https://download.pytorch.org/whl/torch_stable.html
+
+# ORT Module
+RUN pip install --pre onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_torch190.cu102.html
+
+RUN pip install torch-ort
+
```
Comment (thiagocrepaldi) [inline docker/Dockerfile.ort-torch190-onnxruntime-nightly-cu102-cudnn7-devel-ubuntu18.04:33]:
```suggestion
RUN python -m torch_ort.configure

```

File: docker/Dockerfile.ort-torch190-onnxruntime-nightly-cu111-cudnn8-devel-ubuntu18.04 [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+# PyTorch
+RUN pip install onnx ninja
+RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION}+${TORCH_CUDA_VERSION} -f https://download.pytorch.org/whl/torch_stable.html
+
+# ORT Module
+RUN pip install --pre onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_torch190.cu111.html
+
+RUN pip install torch-ort
+
```
Comment (thiagocrepaldi) [inline docker/Dockerfile.ort-torch190-onnxruntime-nightly-cu111-cudnn8-devel-ubuntu18.04:33]:
```suggestion
RUN python -m torch_ort.configure

```

File: docker/Dockerfile.ort-torch190-onnxruntime181-cu102-cudnn7-devel-ubuntu18.04 [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+# PyTorch
+RUN pip install onnx ninja
+RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION}+${TORCH_CUDA_VERSION} -f https://download.pytorch.org/whl/torch_stable.html
+
+# ORT Module
+RUN pip install onnxruntime-training==1.8.1 -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_stable_torch190.cu102.html
+
+RUN pip install torch-ort
+
```
Comment (thiagocrepaldi) [inline docker/Dockerfile.ort-torch190-onnxruntime181-cu102-cudnn7-devel-ubuntu18.04:32]:
```suggestion
RUN python -m torch_ort.configure

```

File: docker/Dockerfile.ort-torch190-onnxruntime181-cu111-cudnn8-devel-ubuntu18.04 [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+# PyTorch
+RUN pip install onnx ninja
+RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION}+${TORCH_CUDA_VERSION} -f https://download.pytorch.org/whl/torch_stable.html
+
+# ORT Module
+RUN pip install onnxruntime-training==1.8.1 -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_stable_torch190.cu111.html
+
+RUN pip install torch-ort
+
```
Comment (thiagocrepaldi) [inline docker/Dockerfile.ort-torch190-onnxruntime181-cu111-cudnn8-devel-ubuntu18.04:32]:
```suggestion
RUN python -m torch_ort.configure

```

File: docker/Dockerfile.ort-torch-and-onnxruntime-nightly-cu102-cudnn7-devel-ubuntu18.04 [hunk 11]
```diff
@@ -26,14 +19,10 @@ RUN pip install --upgrade pip
...
-# RUN pip install torch==1.8.1+cu102 torchvision==0.9.1+cu102 torchaudio==0.8.1 -f https://download.pytorch.org/whl/torch_stable.html
 RUN pip install --pre torch -f https://download.pytorch.org/whl/nightly/cu102/torch_nightly.html
 
 # ORT Module
-# Uncomment one of the lines below. First one gets stable version. Second gets nightly build.
-# RUN pip install onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_stable_cu102.html
-RUN pip install --pre onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_cu102.html
+RUN pip install --pre onnxruntime-training -f https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_torch190.cu102.html
 
 RUN pip install torch-ort
```
Comment (thiagocrepaldi) [inline docker/Dockerfile.ort-torch-and-onnxruntime-nightly-cu102-cudnn7-devel-ubuntu18.04:27]:
```suggestion
RUN pip install torch-ort
RUN python -m torch_ort.configure

```

**Review by thiagocrepaldi — APPROVED** [review]

**Review by harshithapv — APPROVED** [review]

**Pull request merged**